Sources/MapboxMaps/Foundation/Camera/CameraTransition.swift

```diff
@@ -0,0 +1,120 @@
+import UIKit
+import CoreLocation
+
+/// Structure used to represent a desired change to the map's camera
+public struct CameraTransition {
+
+    /// Represents a change to the center coordinate of the map.
+    /// NOTE: Setting the `toValue` of `center` overrides any `anchor` animations
+    public var center: Change<CLLocationCoordinate2D>
+
+    /// Represents a change to the zoom of the map.
+    public var zoom: Change<CGFloat>
+
+    /// Represetns a change to the padding of the map.
+    public var padding: Change<UIEdgeInsets>
+
+    /// Represents a change to the anchor of the map
+    /// NOTE: Incompatible with concurrent center animations
+    public var anchor: Change<CGPoint>
+
+    /// Set this value in order to make bearing/zoom animations
+    /// honor a constant anchor throughout the transition
+    /// NOTE: Incompatible with concurrent center animations
+    public var constantAnchor: CGPoint? {
+        set {
+            anchor.fromValue = newValue ?? anchor.fromValue
+            anchor.toValue = newValue
+        }
+        get {
+            return anchor.fromValue == anchor.toValue ? anchor.fromValue : nil
+        }
+    }
```

Comment:
The more I think about this, the less convinced I am that the value this adds is solving an actual problem. I'm leaning towards removing it and seeing whether people actually have any issues with this API in its absence.

Reply:
Hmm, the call site for an animation with a constant anchor looks strange to me if we don't have something like this:
```swift
mapView.cameraManager.makeAnimator(duration: 3, curve: .linear) { (transition) in 
    transition.bearing.toValue = 90
    transition.anchor.fromValue = somePoint
    transition.anchor.toValue = somePoint
}
```
versus this approach: 

```swift
mapView.cameraManager.makeAnimator(duration: 3, curve: .linear) { (transition) in 
    transition.bearing.toValue = 90
    transition.constantAnchor = somePoint
}
```
What would perhaps be even better is if we could couple the transition that an anchor affects (i.e. bearing changes, zoom changes) with a constant anchor.. Maybe the API could look like:

```swift
mapView.cameraManager.makeAnimator(duration: 3, curve: .linear) { (transition) in 
    transition.bearing.toValue = 90
    transition.bearing.aroundAnchor = somePoint
}
```
But this makes it feel like you could optionally transition around two different anchors for zoom / bearing.. Which is something we can't support.